blob/blob.go

```diff
@@ -67,10 +70,10 @@ type Blob struct {
 // NewBlob constructs a new blob from the provided namespace.ID and data.
 func NewBlob(shareVersion uint8, namespace namespace.ID, data []byte) (*Blob, error) {
 	if len(namespace) != appns.NamespaceSize {
-		return nil, fmt.Errorf("invalid size of the namespace id. got:%d, want:%d", len(nID), appns.NamespaceSize)
+		return nil, fmt.Errorf("invalid size of the namespace id. got:%d, want:%d", len(namespace), appns.NamespaceSize)
 	}
 
-	ns, err := appns.New(namespace[0], namespace[1:])
+	ns, err := appns.New(namespace[versionIndex], namespace[versionIndex+1:])
```

Comment:
It can also be `NamespaceVersionSize-1` and `NamespaceVersionSize`, so that we always up to date with size changes(unlikely but its nice to all the things in sync)